src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -492,28 +494,42 @@ export async function makeUtxoEngine(
     },
 
     async sweepPrivateKeys(spendInfo: EdgeSpendInfo): Promise<EdgeTransaction> {
-      const privateKeys = spendInfo.privateKeys ?? []
-      const dummyNumbWalletInfo: NumbWalletInfo = {
+      // Create the promise to return
+      let success: (
+        value: EdgeTransaction | PromiseLike<EdgeTransaction>
+      ) => void
+      let failure: (err?: any) => void
+      const end = new Promise<EdgeTransaction>((resolve, reject) => {
+        success = resolve
+        failure = reject
+      })
+
+      // The temporary wallet info should include all formats for the currency
+      const allFormats = engineInfo.formats
+      if (allFormats == null) {
+        throw new Error(
+          `Missing  formats for wallet type ${currencyInfo.walletType}`
+        )
+      }
+
+      const tmpWalletInfo: NumbWalletInfo = {
         id: walletInfo.id,
         type: walletInfo.type,
         keys: {
           format: walletInfo.keys.format,
-          supportedFormats: walletInfo.keys.supportedFormats,
+          supportedFormats: allFormats,
           publicKey: {
             publicKeys: {}
           }
         }
       }
+
+      const privateKeys = spendInfo.privateKeys ?? []
       if (privateKeys.length < 1) throw new Error('No private keys given')
-      let success: (
-        value: EdgeTransaction | PromiseLike<EdgeTransaction>
-      ) => void
-      let failure: (reason?: never) => void
-      const end: Promise<EdgeTransaction> = new Promise((resolve, reject) => {
-        success = resolve
-        failure = reject
-      })
-      const tmpDisklet = walletLocalDisklet
+
+      // Make temporary wallet disklet
+      const tmpDisklet = makeMemoryDisklet()
```

Comment:
No problem here. I actually wanted to say that this is very clever and I really like it :)

Reply:
Thanks! 👍